tests/system/source/main.d

```diff
@@ -88,6 +114,143 @@ void main ()
+
+        const Block[] blocks_2 = clients[0].getBlocksFrom(0,3);
+        assert(blocks_2[2].header.height == 2);
+
+        foreach (idx; 0 .. Block.TxsInBlock)
+        {
+            Transaction tx = {
+                type: TxType.Payment,
+                inputs: [Input(hashFull(payment_txs[0]), idx)],
+                outputs: [Output(Amount.MinFreezeAmount, kp.address)]
+            };
+
+            auto signature = kp.secret.sign(hashFull(tx)[]);
+            tx.inputs[0].signature = signature;
+            clients[2].putTransaction(tx);
+        }
+
+        checkBlockHeight(3);
+    }
+
+    /// Create enrollments
+    {
+        API[NodeCnt] clients;
+        foreach (idx, const ref addr; Addrs)
+            clients[idx] = new RestInterfaceClient!API(addr.withSchema());
+
+        Hash[] utxo_hashs;
+        Pair[] signature_noises;
+        Pair[] node_key_pairs;
+        Enrollment[] enrolls;
+
+        foreach (idx; 0 .. NodeCnt)
+        {
+            utxo_hashs ~= UTXOSet.getHash(hashFull(freeze_txs[idx]),0);
+            signature_noises ~= Pair.random;
+
+            Pair node_key_pair;
+            node_key_pair.v = secretKeyToCurveScalar(keypairs[idx].secret);
+            node_key_pair.V = node_key_pair.v.toPoint();
+            node_key_pairs ~= node_key_pair;
+
+            Enrollment enroll;
+            enroll.utxo_key = utxo_hashs[idx];
+            enroll.random_seed = hashFull(Scalar.random());
+            enroll.cycle_length = 1008;
+            enroll.enroll_sig = sign(node_key_pair.v, node_key_pair.V,
+                signature_noises[idx].V, signature_noises[idx].v, enroll);
+            enrolls ~= enroll;
+
+            clients[0].enrollValidator(enrolls[idx]);
+        }
+
+        Thread.sleep(500.msecs);
+
+        // Checking Enrolled
+        size_t times; // Number of times we slept for 50 msecs
+        foreach (idx, ref client; clients)
+        {
+            foreach (utxo_hash; utxo_hashs)
+            {
+                assert(client.hasEnrollment(utxo_hash));
+                writefln("[%s] hasEnrollment: %s", idx, utxo_hash);
+            }
+        }
+    }
+
+    /// Make Transactions for block height 4
+    {
+        API[NodeCnt] clients;
+        foreach (idx, const ref addr; Addrs)
+            clients[idx] = new RestInterfaceClient!API(addr.withSchema());
+
+        const Block[] blocks_3 = clients[0].getBlocksFrom(0,4);
+        assert(blocks_3[3].header.height == 3);
+
+        foreach (idx; 0 .. Block.TxsInBlock)
+        {
+            Transaction tx = {
+                type: TxType.Payment,
+                inputs: [Input(hashFull(blocks_3[3].txs[idx]), 0)],
+                outputs: [Output(Amount.MinFreezeAmount, kp.address)]
+            };
+
+            auto signature = kp.secret.sign(hashFull(tx)[]);
+            tx.inputs[0].signature = signature;
+            clients[2].putTransaction(tx);
+        }
+
+        checkBlockHeight(4);
```

Comment:
You should check that the Enrollments are in the block? Otherwise this entire test doesn't serve any real purpose.

Reply:
I added that check the enrollments are stored in the block